docs/README.md

```diff
@@ -0,0 +1,27 @@
+# Overview
+
+This document provides a high-level overview of the XHarness architecture, which is organized into two main layers: the tooling layer and the application layer.
+
+## Tooling layer
+
+The tooling layer represents the foundation of the tool.
+
+**XHarness.CLI** is the CLI that defines commands and arguments for interacting with XHarness. It serves as the main entry point for users.
```

Comment:
nit: It would be useful to add a note about different commands and supported platforms (e.g., this overview does not mention wasm/wasi support).